app/experimenter/experiments/api/v5/serializers.py

```diff
@@ -219,6 +219,17 @@ class Meta:
         model = NimbusExperiment
         fields = ("probe_sets",)
 
+    def update(self, experiment, data):
+        probe_sets = data.pop("probe_sets")
+        experiment = super().update(experiment, data)
+        with transaction.atomic():
```

Comment:
Could move the experiment update call into the transaction so it all fails as one?